### DISCUSSION TITLE
MonitorControl v4.0.1

### DISCUSSION THREAD

**Discussion by JoniVR** [MonitorControl/MonitorControl#773] [Release Discussion]
This is a service release to address some issues that came up after releasing v4.0.0

### Improvements

- Smoother sliders (asynchronous thread-safe debouncing DDC write)
- Improved support for [BetterDummy](https://github.com/waydabber/BetterDummy)
- Better support for common physical dummies identifying as `28E850`
- Inert dummy menu sliders are now hidden
- Improved support for non-gamma->non-gamma mirroring scenarios
- Add <kbd>command</kbd> + <kbd>w</kbd> key equivalent to "Preferences..." - thanks to @Takuro-Ito
- Compiled to run on macOS 10.14 Mojave as well (use "as is", not supported by the dev team)

### Fixes

- Display properties reset turns off hardware DDC
- Brief black screen upon changing space when using shade dimming
- Text wrap problems on some languages
- Issues with Italian and German translations - thanks to @jajoho and @marcofucito

<hr /><em>This discussion was created from the release <a href='https://github.com/MonitorControl/MonitorControl/releases/tag/v4.0.1'>MonitorControl v4.0.1</a>.</em>

**Comment by wesleycacciari** [comment]
Hi! Great app! I have an 27" Apple Cinema Display (which display brightness adjustment is not working after Monterey Upgrade). I've tried MonitorControl, but it only serve to decrease brightness ... the maximum brightness in this app starts from 50% that the monitor is stuck

**Reply by waydabber** [reply]
Hi, try this:

https://github.com/MonitorControl/MonitorControl/discussions/739

**Comment by wesleycacciari** [comment]
It worked. Thank you!

**Comment by nash1993** [comment]
I've a LG ultrawide (29WN600) monitor. I use with the new M1 macbook air 2020. But everytime I try to change the brightness or sound the monitor restart going black screen for 1-2 seconds. The new update 4.0.1 didn't fix it. I connected the monitor with HDMI.

**Reply by waydabber** [reply]
Hi, this might be an issue with the USB-C to HDMI adapter or the display itself.

**Reply by nash1993** [reply]
The display is fine. May be the issue is i the hub. I'll try another dongle to be sure.

**Reply by waydabber** [reply]
Ok. You might want to try to use USB-C to DisplayPort connectors, those usually work better. USB-C to HDMI dongles are a bit convoluted since most dongles use DisplayPort alt mode internally and a designated converter chip converts the signals to HDMI, including (if the chip is well designed) the i2c bus communication required for DDC control. USB-C to DisplayPort dongles/cables are more straightforward in this regard.

**Reply by waydabber** [reply]
Also note this thread with an other 29" LG user experiencing a similar issue:

https://github.com/MonitorControl/MonitorControl/discussions/760

**Reply by nash1993** [reply]
First let me tell you thanks for this. #760 fixed this bug for brightness. I disabled the hardware DDC and it works. But what to do for sound? The sound slider isn't showing anymore.

**Reply by nash1993** [reply]
Ok problem solved. I manage to control my sound with a software called "eqMac". It works perfectly.

**Reply by waydabber** [reply]
All right. If DDC control is disabled, then the volume cannot be controlled using the mac. MonitorControl changes the volume of the Monitor's amplifier itself by issuing commands to the Monitor, while eqMac changes the intensity of the digital output stream.

**Reply by nash1993** [reply]
I don't know what exactly is the difference technically but in normal usages it works fine. May be proffessionals face problem on it but for now it's alright.

**Reply by waydabber** [reply]
yes, for normal use it's perfectly fine, the end result is basically the same. :)

**Comment by Muirium** [comment]
Using this app every day—on my 4K desktop display and just on my M1 Air when mobile—and still loving it!

A little bug report with this version. I'm running v 4.0.1 build 6941

About half the time I wake my M1 MacBook Air running Monterey 12.0.1, using it in laptop mode with no other displays connected, when I press the brightness keys, I see this:

![Screenshot 2021-11-08 at 10 55 54 pm](https://user-images.githubusercontent.com/221646/140832082-fa3223ff-a9a2-4d6b-b794-dafb9f30d9d5.jpg)

That icon means my controls are blocked. The display will stay at the same setting no matter what I do when it shows up.

Sleep / waking the Mac again fixes this. Sleep / waking the Mac once again often brings it back!

Trying to trigger it in clamshell mode with my external 4K display just now is proving more difficult. I'm pressing brightness keys on my USB keyboard but nothing happens, not even the UI in the screenshot above. MonitorControl's sliders still work though with a trackpad.

Testing with both displays visible (not easy with my setup, have to hold the Mac with one hand while doing this…) I can confirm the same graphic appears on both displays. Brightness control seems to be restored after about 10 seconds of repeated button pressing there.

Here's my settings, in case anything is relevant:

![Screenshot 2021-11-08 at 11 16 05 pm](https://user-images.githubusercontent.com/221646/140833277-8120c1d9-276b-477d-b31e-98420f514a58.jpg)
![Screenshot 2021-11-08 at 10 58 23 pm](https://user-images.githubusercontent.com/221646/140833281-bc1931d7-0030-40de-9066-c750dc4ea1fd.jpg)
![Screenshot 2021-11-08 at 10 58 17 pm](https://user-images.githubusercontent.com/221646/140833283-b25f53dc-8d3a-485b-98d8-5c1acd0d3090.jpg)
![Screenshot 2021-11-08 at 10 57 54 pm](https://user-images.githubusercontent.com/221646/140833287-172143d3-4ae0-46ea-8dc8-d705c0eaec6a.jpg)
![Screenshot 2021-11-08 at 10 57 50 pm](https://user-images.githubusercontent.com/221646/140833289-49066c8c-0cf2-4f95-a6cf-944bac4f0c3a.jpg)

**Reply by waydabber** [reply]
Hi @Muirium 

thanks for the feedback!

Whenever the display is reconfigured or after wakes up, there is a 5-8 second long period (depending whether the reconfiguration was happening due to sleep or display connect/disconnect) when controls are not available. The locked symbol signifies to the user that changing brightness or volume is not possible during this period. This is implemented because there are some displays which may take time to wake up properly (some older displays have a really slow firmware or need time to fire up the backlighting system) and especially older/slower macs may work on display reconfiguration after a change for extended periods of time, during which issuing DDC commands or altering the display or creating shades for some configurations are detrimental.

Opening/closing the lid, connecting/disconnecting the display or sleeping/waking the mac again actually restarts the timer so instead of speeding up/helping the process, slows it down. :) All you need to do is to wait a bit after sleep or connecting the display and things should work.

We might add a setting in the future to change the default timings for users with faster displays/macs (you can add a feature request for this).

Hope this helps! :) If you still feel that there is an issue (like the app never recovers from the configuration state), we can try to figure that out.

**Reply by Muirium** [reply]
Ah! Understood. It just looked like a bug to me, not expected behaviour.

I encounter the lock-out much more frequently on my Air's built in display than when docked at my desk. My Dell takes many seconds to fully power up, so usually skips that time-out. (As is likely intended.)

But my Air wakes much quicker! Being locked out of Brightness control on wake is really quite annoying, as that's when I'm most likely to need to change it. A setting to avoid this lock-out on an M1's internal display would be very helpful indeed!

Is anyone on Intel MacBooks noticing it?

**Reply by waydabber** [reply]
Yes, for MacBook displays only this feature would not be needed as there is no lag/slow wake issue with that. You might want to add an issue about it, thie way this works might be modified.

**Comment by stevenlynch12** [comment]
I have skipped multiple versions so I am not sure when this issue came up. In the latest version, mute/turning the volume all the way down is not muted. It is a very low volume. When I use the volume slider in the menu bar it shows it at zero percent but I can still hear. With my Samsung C34J79x monitor, the lowest volume is not very low. I am running Mac OS Moneray. Not sure if I am missing something in the setting for this.

**Reply by waydabber** [reply]
This is a display specific issue. You might want to try `Enable Mute DDC Command` under Displays, sometimes this helps displays which do not fully mute the speaker on the 0 volume command.

<img width="842" alt="Screen Shot 2021-11-09 at 10 15 46" src="https://user-images.githubusercontent.com/37590873/140896340-bcb88101-8a62-4df2-a7b8-851745ec2055.png">

Hope this helps!

**Comment by riot1976** [comment]
Thank for the great tool (Imac27 2020 and Imac late 2009 as second screen)! Just one question... is there an opportunity in the tool to (auto) reduce the brightness (like a soft sleep mode) just for the second screen? For example, if the mouse curser was 5 minutes not on the second screen. I am working just partly with this second screen (Imac 2009) e.g video calls etc. but the screen is almost the whole day on (energy consumption, dazzle effect..) Thank you!

**Reply by waydabber** [reply]
A feature like this could be added in the future - I recommend you opening a new Feature Request for it under Issues. :)
